sharding/nodesCoordinator/sovereignIndexHashedNodesCoordinator.go

```diff
@@ -62,6 +62,7 @@ func NewSovereignIndexHashedNodesCoordinator(arguments ArgNodesCoordinator) (*so
 	ihnc.loadingFromDisk.Store(false)
 
 	ihnc.nodesCoordinatorHelper = ihnc
+	ihnc.numberOfShardsComputer = ihnc
```

Comment:
Isn't this approach a bit "tricky"?
You assign an internal interface to yourself (your own behavior).

Why not use the usual approach and have this as an input interface parameter?